backend/kernelCI_app/management/commands/helpers/kcidbng_ingester.py

```diff
@@ -277,33 +312,167 @@ def db_worker(stop_event: threading.Event):
+                    )
+                    _out(
+                        "[%s] bulk_create builds: n=%d in %.3fs"
+                        % (_ts(), len(builds_buf), time.time() - t0)
+                    )
+                if tests_buf:
+                    t0 = time.time()
+                    Tests.objects.bulk_create(
+                        tests_buf, batch_size=batch_size, ignore_conflicts=True
+                    )
+                    _out(
+                        "[%s] bulk_create tests: n=%d in %.3fs"
+                        % (_ts(), len(tests_buf), time.time() - t0)
+                    )
+                if incidents_buf:
+                    t0 = time.time()
+                    Incidents.objects.bulk_create(
+                        incidents_buf, batch_size=batch_size, ignore_conflicts=True
+                    )
+                    _out(
+                        "[%s] bulk_create incidents: n=%d in %.3fs"
+                        % (_ts(), len(incidents_buf), time.time() - t0)
+                    )
+        except Exception as e:
+            logger.error("Error during bulk_create flush: %s", e)
+        finally:
+            flush_dur = time.time() - flush_start
+            rate = total / flush_dur if flush_dur > 0 else 0.0
+            msg = (
+                "[%s] Flushed batch in %.3fs (%.1f items/s): "
+                "issues=%d checkouts=%d builds=%d tests=%d incidents=%d"
+                % (
+                    _ts(),
+                    flush_dur,
+                    rate,
+                    len(issues_buf),
+                    len(checkouts_buf),
+                    len(builds_buf),
+                    len(tests_buf),
+                    len(incidents_buf),
+                )
+            )
+            _out(msg)
+            issues_buf.clear()
+            checkouts_buf.clear()
+            builds_buf.clear()
+            tests_buf.clear()
+            incidents_buf.clear()
+            last_flush_ts = time.time()
+
     while not stop_event.is_set() or not db_queue.empty():
         try:
             item = db_queue.get(timeout=0.1)
             if item is None:
-                db_queue.task_done()  # Important: mark the poison pill as done
+                db_queue.task_done()
                 break
             try:
                 data, metadata = item
                 if data is not None:
-                    process_submission_item(data, metadata)
+                    inst = build_instances_from_submission(data)
+                    issues_buf.extend(inst["issues"])
+                    checkouts_buf.extend(inst["checkouts"])
+                    builds_buf.extend(inst["builds"])
+                    tests_buf.extend(inst["tests"])
+                    incidents_buf.extend(inst["incidents"])
+
+                # Flush if batch size reached
+                if buffered_total() >= INGEST_BATCH_SIZE:
+                    flush_buffers()
```

Comment:
nit: no need for comments like this when the code is clear
```suggestion
                if buffered_total() >= INGEST_BATCH_SIZE:
                    flush_buffers()
```